crates/binjs_io/src/multipart/write.rs

```diff
@@ -55,25 +56,35 @@ impl Serializable for Vec<u8> {
 }
 
 /// A `String` is serialized as:
-/// - number of UTF-8 bytes (varnum);
-/// - sequence of UTF-8 bytes.
+/// - number of WTF-8 bytes (varnum);
+/// - sequence of WTF-8 bytes.
 impl Serializable for SharedString {
     fn write<W: Write>(&self, out: &mut W) -> Result<usize, std::io::Error> {
         let mut total = 0;
-        total += out.write_varnum(self.len() as u32)?;
-        out.write_all(self.deref().as_bytes())?;
+        let bytes = self.deref().as_bytes();
+
+        match escaped_wtf8::unescape(bytes) {
```

Comment:
Since you’re writing a generic `W: fmt::Write` stream, maybe `unescape` could be generic and take `&mut W` as well to avoid allocating at all?

Reply:
do you mean changing the unescape function to unescape_and_write or something?

Reply:
Maybe you’d want to rename it at the same time, but I meant that this `&mut W` parameter would be instead of the return value, not in addition to.

However that’s perhaps a more disruptive change than is worth doing speculatively before profiling to see if those memory allocation take any significant time at all.